### PR TITLE
Feature/sign out

### PR DESCRIPTION
**Pull request by t-romani**

### CODE REVIEW

File: spec/factories/user.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FactoryBot.define do
+  factory :user do
+    email     { 'test_user@example.com' }
```
Comment (gonzaM) [inline spec/factories/user.rb]: You can use `Faker::Internet.email` here

File: spec/factories/user.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    password  { Faker::Internet.password }
+    full_name { Faker::Name.name_with_middle }
+    gender    { 2 }
```
Comment (gonzaM) [inline spec/factories/user.rb]: You can use something like: `User.genders.keys.sample` (.sample returns a random element from the array)

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -1,13 +1,11 @@
 require 'rails_helper'
-RSpec.describe User, :type => :model do
-  subject do
-    @user1 = FactoryBot.create(:user)
-  end
+describe User, :type => :model do
```
Comment (gonzaM) [inline spec/models/user_spec.rb]: Use the new syntax: `type: :model`

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
+  subject { @user1 = create(:user) }
 
   describe '#valid attributes' do
     it 'is valid with valid attributes' do
```
Comment (gonzaM) [inline spec/models/user_spec.rb]: The description should be `'is valid'` then you can use a context '`with valid params`'. So `with valid params... is valid`

File: spec/request/session/create_spec.rb [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
-    post '/api/v1/users/sign_in', params: {
-      email: user_attrs[:user][:email], password: user_attrs[:user][:password]
+    post api_v1_user_session_path, params: {
+      email: user_attrs[:user][:email], password: password
     }, as: :json
   end
 
   context 'when valid' do
+    let(:password) { 'example' }
     context 'when registered and confirmed' do
```
Comment (gonzaM) [inline spec/request/session/create_spec.rb:13]: Add an empty line after the `let`

File: spec/request/session/create_spec.rb [hunk 6]
```diff
@@ -26,7 +27,8 @@
 
   context 'when invalid' do
     context 'when registered and not confirmed' do
-      let!(:user) { FactoryBot.create(:user) }
+      let(:password) { 'example' }
+      let!(:user) { create(:user, password: password) }
```
Comment (gonzaM) [inline spec/request/session/create_spec.rb]: I would align this

File: spec/request/session/create_spec.rb [hunk 7]
```diff
@@ -39,19 +41,16 @@
     end
 
     context 'when registered and confirmed but invalid credentials' do
-      let!(:user) { FactoryBot.create(:confirmed_user) }
+      let!(:user) { create(:confirmed_user) }
+      let(:password) { 'another' }
```
Comment (gonzaM) [inline spec/request/session/create_spec.rb]: Same as above (align)

File: spec/request/session/delete_spec.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+describe 'POST #delete sign_out', type: :request do
+  let!(:user_attrs) { { user: attributes_for(:user) } }
+  let!(:user) { create(:user, :confirmed) }
+  let!(:auth_headers) { auth_user_headers }
```
Comment (gonzaM) [inline spec/request/session/delete_spec.rb]: I would align this too and also add an empty line after let

File: spec/request/session/delete_spec.rb [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  subject do
+    delete destroy_api_v1_user_session_path, headers: auth_headers, as: :json
+  end
+
+  context 'when valid' do
+    context 'when signed in' do
```
Comment (gonzaM) [inline spec/request/session/delete_spec.rb:12]: There is no context for `when not signed in`?

File: spec/request/session/delete_spec.rb [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+      it 'does get a sucessful answer' do
```
Comment (gonzaM) [inline spec/request/session/delete_spec.rb]: just `gets a ...`? Same for all descriptions

File: spec/request/session/delete_spec.rb [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+        expect { subject }.to change { User.last.tokens.size }.by(-1)
+      end
+    end
+  end
+
+  context 'when invalid' do
+    context 'when not signed in' do
+      it 'does get a not found error' do
+        subject
+        delete destroy_api_v1_user_session_path, headers: auth_headers, as: :json
```
Comment (gonzaM) [inline spec/request/session/delete_spec.rb]: I don't think this is the best way to do this. Use a `before` block if necessary. Or even better, don't log in the user.

File: spec/request/user/create_spec.rb [hunk 12]
```diff
@@ -56,7 +56,7 @@
 
     context 'when invalid password' do
       it '(different) it does not create a new user' do
-        user_params[:user][:password] = 'password'
+        # user_params[:user][:password] = 'password'
```
Comment (gonzaM) [inline spec/request/user/create_spec.rb]: Remove commented code

**Review by gonzaM — APPROVED** [review]
Good job!

File: spec/request/session/delete_spec.rb (hunk 9 above)
Comment (t-romani) [reply spec/request/session/delete_spec.rb:12]: there is further on, inside the invalid context

**Review by Alxzu — APPROVED** [review]

**Review by JuanfraM — APPROVED** [review]
good job!

**Pull request merged**